pkg/balancer/vhost_server.go

```diff
@@ -0,0 +1,96 @@
+package balancer
+
+import (
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	vhost "github.com/inconshreveable/go-vhost"
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/util"
+)
+
+// VhostServer uses TLS virtual hosting using SNI to direct requests to
+// an appropriate backend.
+type VhostServer struct {
```

Comment:
looks like you use the constructor though? i assume you mean bin/p2-balancer/main.go?